search_knn.go

```diff
@@ -168,7 +168,8 @@ var (
 )
 
 func queryWithKNN(req *SearchRequest) (query.Query, error) {
-	if len(req.KNN) > 0 {
+	_, ok := req.Query.(*query.KNNQuery)
```

Comment:
We do support KNNQueries when using bleve as a library. Example code snippet:
```
nestedKNNQuery := query.NewKNNQuery(getQueryVec("hilly region worldwide"))
nestedKNNQuery.SetFieldVal("vector")
nestedKNNQuery.SetK(3)
```
This results in a KNN query which then needs to be added to the KNN field of the search request(done in queryWithKNN()).

Reply:
But if we are going to work with KNNs using existing code for conj/disj *queries*, we need to support kNN queries since conjuncts/disjuncts are of type Query.

Reply:
As per offline conversation, this condition is necessary only when the query is a KNN query, which we don't support since the users should populate the KNN field instead of query.